### PR TITLE
Games (connect4, blackjack)

### PR DESCRIPTION
**Pull request by Faholan**
I commented out all database-related code, and added # DATABASE at the 
end of each line I had to add because of that

### CODE REVIEW

**Comment by Faholan** [comment]
This PR requires :
git+https://github.com/Rapptz/discord-ext-menus.git

**Comment by Theodore-Robinson** [comment]
> This PR requires :
> git+https://github.com/Rapptz/discord-ext-menus.git

Why do we?

File: cogs/games.py [hunk 1]
```diff
@@ -33,10 +36,436 @@
 ]
 
 
+class connect4(menus.Menu):
```
Comment (ItsDrike) [inline cogs/games.py:39]: Classes should start with uppercase first letter.
```suggestion
class Connect4(menus.Menu):
```
(Note, this also means you'll need to change all references to it)

File: cogs/games.py [hunk 2]
```diff
@@ -33,10 +36,436 @@
...
         self.bot = bot
 
+    @command(ignore_extra=True)
+    async def blackjack(self, ctx: Context, cost: int = 5) -> None:
+        """Please see the detailed help
+        Rules : if it's your turn, press the button corresponding to the column in which you wana place the card.
+        If you want to split (play on one more column, up to a max of 3, press :regional_indicator_3:). If you want to stop, press :x:.
+        To win, you must score more than the dealer, but no more than 21 (each card's value is its pip value, except faces, which are worth 10 \
+        points, and the Ace, which is worth either 1 or 11).
+        An Ace plus a face is called a blackjack, and beats a 21"""
```
Comment (ItsDrike) [inline cogs/games.py:452]: Docstring formatting adjustment.
```suggestion
        """
        Rules: if it's your turn, press the button corresponding to the column in which you want to place the card.
        If you want to split (play on one more column, up to a max of 3, press :regional_indicator_3:).  If you want to stop, press :x:.
        To win, you must score more than the dealer, but no more than 21 (each card's value is its pip value, 
        except for faces, which are worth 10 points, and the Ace, which is worth either 1 or 11).
        An Ace plus a face is called a blackjack and beats a 21
        """
```

File: cogs/games.py [hunk 3]
```diff
@@ -33,10 +36,436 @@
...
+        for card in self:
+            if card.is_ace:
+                c += 1
+        while c:
+            if V <= 11:
+                V += 10
+                c -= 1
+            else:
+                break
+        return V
```
Comment (ItsDrike) [inline cogs/games.py:192]: I'm not a huge fan of these undescriptive 1-letter variables

**Review by ItsDrike — CHANGES_REQUESTED** [review]
Nice work, just a few changes. 
But we should use our paginator instead of the ext-menus as Theodore-Robinson pointed out.

**Comment by Faholan** [comment]
> 
> 
> > This PR requires :
> > git+https://github.com/Rapptz/discord-ext-menus.git
> 
> Why do we?
> We already have a paginator that's under active development on [pr 43](https://github.com/The-Codin-Hole/HotWired-Bot/pull/43)

The thing is, your paginator is based on pages. But I don't need pages : what I need, is a way to run functions on input of a given emoji. If I chose your paginator, I'd need to override many methods. But discord.ext.menus allows me to only override a few methods, because it was made to give people the ability to choose exactly what reaction does what, with just a decorator. It's much easier

**Comment by Theodore-Robinson** [comment]
> > > This PR requires :
> > > git+https://github.com/Rapptz/discord-ext-menus.git
> > 
> > 
> > Why do we?
> > We already have a paginator that's under active development on [pr 43](https://github.com/The-Codin-Hole/HotWired-Bot/pull/43)
> 
> The thing is, your paginator is based on pages. But I don't need pages : what I need, is a way to run functions on input of a given emoji. If I chose your paginator, I'd need to override many methods. But discord.ext.menus allows me to only override a few methods, because it was made to give people the ability to choose exactly what reaction does what, with just a decorator. It's much easier

Yeah, I understand what you mean but the paginator does not just give you the utility to work with pages, it calls many functions that let you run specific parts such as the reaction function you discussed. For example, the paginator that is being worked on can do this:
```py
from utils/paginator import respond_to_reaction

def function1():
    print('Received reaction and invoked function1')

@bot.command()
async def react_to_me(ctx):
    message = await ctx.send('Reacting to this message..')
    respond_to_reaction(ctx, {reaction:'✔️', action: function1})
```

**Comment by Faholan** [comment]
> 
> 
> > > > This PR requires :
> > > > git+https://github.com/Rapptz/discord-ext-menus.git
> > > 
> > > 
> > > Why do we?
> > > We already have a paginator that's under active development on [pr 43](https://github.com/The-Codin-Hole/HotWired-Bot/pull/43)
> > 
> > 
> > The thing is, your paginator is based on pages. But I don't need pages : what I need, is a way to run functions on input of a given emoji. If I chose your paginator, I'd need to override many methods. But discord.ext.menus allows me to only override a few methods, because it was made to give people the ability to choose exactly what reaction does what, with just a decorator. It's much easier
> 
> Yeah, I understand what you mean but the paginator does not just give you the utility to work with pages, it calls many functions that let you run specific parts such as the reaction function you discussed. For example, the paginator that is being worked on can do this:
> 
> ```python
> from utils/paginator import respond_to_reaction
> 
> def function1():
>     print('Received reaction and invoked function1')
> 
> @bot.command()
> async def react_to_me(ctx):
>     message = await ctx.send('Reacting to this message..')
>     respond_to_reaction(ctx, {reaction:'✔️', action: function1})
> ```



> 
> 
> > > > This PR requires :
> > > > git+https://github.com/Rapptz/discord-ext-menus.git
> > > 
> > > 
> > > Why do we?
> > > We already have a paginator that's under active development on [pr 43](https://github.com/The-Codin-Hole/HotWired-Bot/pull/43)
> > 
> > 
> > The thing is, your paginator is based on pages. But I don't need pages : what I need, is a way to run functions on input of a given emoji. If I chose your paginator, I'd need to override many methods. But discord.ext.menus allows me to only override a few methods, because it was made to give people the ability to choose exactly what reaction does what, with just a decorator. It's much easier
> 
> Yeah, I understand what you mean but the paginator does not just give you the utility to work with pages, it calls many functions that let you run specific parts such as the reaction function you discussed. For example, the paginator that is being worked on can do this:
> 
> ```python
> from utils/paginator import respond_to_reaction
> 
> def function1():
>     print('Received reaction and invoked function1')
> 
> @bot.command()
> async def react_to_me(ctx):
>     message = await ctx.send('Reacting to this message..')
>     respond_to_reaction(ctx, {reaction:'✔️', action: function1})
> ```

Honestly, I'd find it really annoying to have to re-write everything : this code works, with a paginator created by the great Rapptz himself. It was already a pain in the arse writing it so that Travis would be pleased, and I'm no masochist wanting to go throught the same process another time just to use a specific paginator.

**Comment by Theodore-Robinson** [comment]
Yes but @Faholan its more maintainable and readable and even more usable if its something we have written and if the paginator requires this function to be made for it, why not use it further instead of using it just once for the paginator?

**Comment by Theodore-Robinson** [comment]
> > Yes but @Faholan its more maintainable and readable and even more usable if its something we have written and if the paginator requires this function to be made for it, why not use it further instead of using it just once for the paginator?
> 
> Two comments : I checked, the code you sent me isn't currently usable (respond_to_reaction isn't defined. If I were to use the current Paginator, I'd have to override both __init__ (because I don't use pages) and send (because I don't use 5 emojis) methods. But the Paginator only has those two methods. So it'd be stupid to use the Paginator.
> 
> And a little tip : I see what you tried to do, and I advise you to go check `discord.ext.menus` : using MenuPages and ListPageSource is much easier. You can see how I use them here : https://github.com/Faholan/All-Hail-Chaos/blob/dc769d5152f73f9c5c3778cea08f071d21d64b81/cogs/utility.py#L40

Didn't I mention its under development?
It will be out for a final release and I'm off to a break.
On the advise for me to use discord.ext.menus, using something built by yourself is better to both maintain and edit.

**Review by janaSunrise — APPROVED** [review]

**Pull request merged**